modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2032,27 +2148,81 @@ def _execute_arrow(self):
         pyarrow.Table
             The resulting table.
         """
-        if isinstance(self._op, FrameNode):
-            if self._partitions.size == 0:
-                return pyarrow.Table.from_pandas(pd.DataFrame({}))
+        result = None
+        stack = [self]
+
+        while stack:
+            frame = stack.pop()
+
+            if callable(frame):
+                result = frame()
+            elif isinstance(frame._op, FrameNode):
+                if frame._partitions.size == 0:
+                    result = pyarrow.Table.from_pandas(pd.DataFrame({}))
```

Comment:
even an empty dataframe may sometime contain some metadata, can we preserve it by doing something like this?
```suggestion
                    result = pyarrow.Table.from_pandas(pd.DataFrame(index=self._index_cache, columns=self._columns_cache))
```